omniscidb/QueryEngine/ExtensionsIR.cpp

```diff
@@ -647,7 +643,8 @@ void CodeGenerator::codegenBufferArgs(const std::string& ext_func_name,
         cgen_state_->ir_builder_.CreateStructGEP(buffer_abstraction, alloc_mem, 2);
     cgen_state_->ir_builder_.CreateStore(buffer_null_extended, buffer_is_null_ptr);
   }
-  output_args.push_back(alloc_mem);
+  output_args.push_back(
+      cgen_state_->ir_builder_.CreateLoad(buffer_abstraction, alloc_mem));
```

Comment:
Why do you replace a pointer to the allocated structure with an actual structure value here?

Reply:
I think it might be a bug. It adds this seemingly useless load:
```
  %14 = load { ptr, i64, i8 }, ptr %buffer_arg_ptr, align 8 
  %15 = alloca { ptr, i64, i8 }, align 8
  call void @array_append__4(ptr %15, ptr %buffer_arg_ptr, float %5) 
```
The load is never used. It disappears in the optimized IR, too. But without it, the array_append function doesn't return properly. The operation is sorta hacked anyway, the struct return can be optimized in different ways depending on the size which is why I opted for an int64 in the middle. I wonder if the opaque ptr type vs the previous 64-bit pointer is the problem? At least I think it was 64-bits.  Anyway, adding the load seemed to resolve the problem.

Reply:
OK, just wanted to make sure this was intentional

Reply:
Yeah, it's covered pretty well by the tests in `ArrayTest`. When I wrote this I knew I was hacking the ABI, and so I made sure there was good test coverage. :)